internal/mode/static/health.go

```diff
@@ -13,37 +20,94 @@ func newGraphBuiltHealthChecker() *graphBuiltHealthChecker {
 	}
 }
 
-// graphBuiltHealthChecker is used to check if the initial graph is built and the NGF Pod is ready.
+// graphBuiltHealthChecker is used to check if the NGF Pod is ready. The NGF Pod is ready if the initial graph has
+// been built and if it is leader.
 type graphBuiltHealthChecker struct {
 	// readyCh is a channel that is initialized in newGraphBuiltHealthChecker and represents if the NGF Pod is ready.
-	readyCh chan struct{}
-	lock    sync.RWMutex
-	ready   bool
+	readyCh    chan struct{}
+	lock       sync.RWMutex
+	graphBuilt bool
+	leader     bool
+}
+
+// createHealthProbe creates a Server runnable to serve as our health and readiness checker.
+func createHealthProbe(cfg config.Config, healthChecker *graphBuiltHealthChecker) (manager.Server, error) {
+	// we chose to create our own health probe server instead of using the controller-runtime one because
+	// of an annoying log which would flood our logs on non-ready non-leader NGF Pods. This health probe is pretty
+	// similar to the controller-runtime's health probe.
```

Comment:
```suggestion
	// we chose to create our own health probe server instead of using the controller-runtime one because
	// of a repetitive log which would flood our logs on non-ready non-leader NGF Pods. This health probe is
	// similar to the controller-runtime's health probe.
```

Making the language a bit more generic.